community/server/src/test/java/org/neo4j/server/rest/transactional/integration/TransactionIT.java

```diff
@@ -585,17 +600,10 @@ public void begin__execute__rollback_concurrently() throws Exception
         } );
 
         // terminate
-        final Future<Response> interruptFuture = Executors.newSingleThreadExecutor().submit( () ->
+        final Future<Response> interruptFuture = executors.submit( () ->
         {
-            try
-            {
-                latch.await();
-                Thread.sleep( 100 );
-            }
-            catch ( InterruptedException ignored )
-            {
-                // go
-            }
+            waitForStatementExecution( statement );
```

Comment:
Looks like we are still racing with the completion of the `statement` query.

I think it would be more robust to have a third transaction lock a node, make the statement try to modify a property of that node where it will get stuck on the entity lock. Then we should be able to wait for the statement execution without worrying that it might have passed us by already.

This would incidentally expand the scope of the test to include verifying that we can terminate transactions even when they are waiting in the lock manager.